[IMP] VoIP: Devices + Integrations Revamp (#13002)
odoo/documentation: content/applications/productivity/voip/devices_integrations.rst

=============================
VoIP devices and integrations
=============================

.. |VOIP| replace:: :abbr:`VoIP (Voice over Internet Protocol)`
.. |SIP| replace:: :abbr:`SIP (Session Initiation Protocol)`

> While |VOIP| is available on Odoo's desktop experience, it can also be used through the Odoo mobile

Reviewer: link doc for more info on mobile apps

app and through third-party apps as well. As long as the app is compatible with |SIP|, it can be
used to make and receive calls with Odoo |VOIP|.

Odoo |VOIP| is fully integrated with the **Contacts**, **CRM**, **Sales**, and **Helpdesk** apps.
While the benefits are most clearly felt when using |VOIP| from the Odoo database, it can also be

> used from apps like the ones covered below.

Reviewer: Highly optional, but I would list the apps rather than use a spatial reference, since doc things move around with time

.. important::
   Using |VOIP| through the Odoo mobile app or through a third-party tool still requires a |VOIP|

>    service provider. Learn more about signing a :doc:`VoIP service provider <../voip>`.

Reviewer: Move to a "see more" admonition or finish the sentence with a prepositional phrase so it's not just a command to learn

VoIP via the Odoo mobile app
============================

Odoo |VOIP| is available through mobile browsers as well as through the Odoo app (available through
the Apple App Store and Google Play Store). Odoo recommends using a mobile browser to access a
database, rather than the mobile apps.

.. danger::
   Odoo mobile apps are no longer supported by Odoo. Learn more about :doc:`Odoo's mobile apps

>    <../../../administration/mobile>`.

Reviewer: Does this contradict the first sentence encouraging you to use mobile apps? I assume this is distinguishing the PWA app from some deprecated one but for the average user this looks like it would be confusing.

Make a call through the mobile app
----------------------------------

To make a call through the Odoo mobile app, first tap the :icon:`oi-voip` :guilabel:`VoIP` icon at
the top of the screen. From here, the |VOIP| widget will appear on the screen. After that, the user
can start a call using one of three ways:

- Enter the phone number to be called by clicking the :icon:`fa-keyboard` :guilabel:`(keyboard)`
  icon, and then entering the phone number.
- Click the :icon:`fa-phone` :guilabel:`(phone)` icon to redial the last called contact.
- Search for a specific contact's name or go to the :guilabel:`Contacts` tab. Then, select the
  contact and click the :icon:`fa-phone` :guilabel:`(phone)` icon.

When the first call from the mobile app is made, a prompt to allow the database to use the device's
microphone appears. When this prompt appears, click :guilabel:`Allow` to continue with the call.

This step is **necessary**, regardless of whether the mobile Odoo application or web browser is

> used.

Reviewer: admonition and specify what it blocks

Odoo then asks how to make the call. The two options are to either use |VOIP| or use the phone's
connection. Click the box next to :guilabel:`Remember ?` should this decision be the default moving
forward.

.. _voip/zoiper:

Zoiper Lite
===========

*Zoiper Lite* third-party alternative to Odoo |VOIP|. It is a free |VOIP| |SIP| dialer with voice
and video.

> Download the *Zoiper Lite* app from either the `Google's Play Store

Reviewer: ```suggestion
Download the *Zoiper Lite* app from either the `Google Play Store
```

<https://play.google.com/store/apps/details?id=com.zoiper.android.app>`_ or `Apple's App Store
<https://apps.apple.com/us/app/zoiper-lite-voip-soft-phone/id438949960>`_ (depending on the device's

> operating system) to begin using *Zoiper Lite*. This article will cover setting up *Zoiper Lite* for

Reviewer: "Download Zoiper... to begin using Zoiper"

Reviewer: Do we refer to doc subsections as "articles"?

an iOS device while using :doc:`Axivox <axivox/axivox_config>` as the |VOIP| service provider, so
the experience may differ for other operating systems or |VOIP| service providers.

> Once *Zoiper Lite* is installed on the mobile device, open the app and tap on :guilabel:`Settings`.

Reviewer: ```suggestion
Once *Zoiper Lite* is installed on the mobile device, open the app and tap :guilabel:`Settings`.
```

> Navigate to :menuselection:`Accounts`, and tap on the :guilabel:`+ (plus)` icon to add an account.

Reviewer: ```suggestion
Navigate to :menuselection:`Accounts`, and tap the :guilabel:`+ (plus)` icon to add an account.
```

If a |VOIP| service provider is signed, then tap :guilabel:`Yes` when the app asks if the user has
an account username and password. From here, tap :guilabel:`Manual configuration` unless the |VOIP|
service provider's country is known (if it is, then tap :guilabel:`Select provider` and find the
provider through the list).

> If :guilabel:`Manual configuration` was tapped, then tap :guilabel:`SIP account`. From here, under

Reviewer: ```suggestion
If :guilabel:`Manual configuration` was tapped, tap :guilabel:`SIP account`. From here, under
```
if...then == causality, not procedure

:guilabel:`SIP OPTIONS`, fill out the fields:

- :guilabel:`Account name`: How this account appears in *Zoiper Lite*. Can be anything.
- :guilabel:`Domain`: The |VOIP| service provider's domain. This is unique to each service provider
  account.
- :guilabel:`Username`: The user's |SIP| username. This is unique to each user account in the
  service provider account.
- :guilabel:`Password`: The user's |SIP| password. This is unique to each user account in the
  service provider account.

.. tip::
   To access this information through the *Axivox* portal, click :guilabel:`Users`, then choose a
   user and click :guilabel:`Edit`, and finally click the :guilabel:`SIP Identifiers` tab.

Once all of this information is entered, tap the green :guilabel:`Register` button at the top of the
screen. *Zoiper* will confirm the information, then show a message saying :guilabel:`Registration
Status: OK` if everything was correct. From here, the *Zoiper Lite* app can now be used to make and
take phone calls using the |VOIP| service.

Linphone
========

> *Linphone* third-party alternative to Odoo |VOIP|. It is a free |VOIP| |SIP| dialer with voice,

Reviewer: ```suggestion
*Linphone* is a third-party alternative to Odoo |VOIP|. It is a free |VOIP| |SIP| dialer with voice,
```

video, messaging (direct messages and group chats), and can support conference calls.

Download the *Linphone* app from either the `Google Play Store
<https://play.google.com/store/apps/details?id=org.linphone>`_ or the `Apple App Store

> <https://apps.apple.com/us/app/linphone/id360065638>`_ (depending on the device's operating system)

Reviewer: You probably don't need to explain that it depends on OS in this day and age

> to begin using *Linphone*. This article will cover setting up *Linphone* for an iOS device while

Reviewer: The article thing again. Like maybe these should just actually be fully separate docs?

   :doc:`Odoo's mobile apps <../../../administration/mobile>`.

Make a call through the mobile app
----------------------------------

To make a call through the Odoo mobile app, first tap the :icon:`oi-voip` :guilabel:`VoIP` icon at
the top of the screen. From here, the |VOIP| widget will appear on the screen. After that, the user
can start a call using one of three ways:

- Enter the phone number to be called by clicking the :icon:`fa-keyboard` :guilabel:`(keyboard)`
  icon, and then entering the phone number.
- Click the :icon:`fa-phone` :guilabel:`(phone)` icon to redial the last called contact.
- Search for a specific contact's name or go to the :guilabel:`Contacts` tab. Then, select the
  contact and click the :icon:`fa-phone` :guilabel:`(phone)` icon.

When the first call from the mobile app is made, a prompt to allow the database to use the device's
microphone appears. When this prompt appears, click :guilabel:`Allow` to continue with the call.

.. important::
   Allowing the app to use the device's microphone is **necessary**, regardless of whether the
   mobile Odoo application or web browser is used. If permission is not granted, calls cannot be
   made from the app.

> Odoo then asks how to make the call. The two options are to either use |VOIP| or use the phone's
> connection. Click the box next to :guilabel:`Remember ?` should this decision be the default moving
> forward.

Reviewer 2: possible rewrite for brevity

```suggestion
Odoo then prompts for a call method: VoIP or the device’s cellular connection. Select Remember? to set the chosen option as the default.
```

Note: Didn't format so remember to do so in your revision!

an account username and password. From here, tap :guilabel:`Manual configuration` unless the |VOIP|
service provider's country is known (if it is, then tap :guilabel:`Select provider` and find the
provider through the list).

If :guilabel:`Manual configuration` was tapped, tap :guilabel:`SIP account`. From here, under
:guilabel:`SIP OPTIONS`, fill out the fields:

- :guilabel:`Account name`: How this account appears in *Zoiper Lite*. Can be anything.
- :guilabel:`Domain`: The |VOIP| service provider's domain. This is unique to each service provider
  account.
- :guilabel:`Username`: The user's |SIP| username. This is unique to each user account in the
  service provider account.
- :guilabel:`Password`: The user's |SIP| password. This is unique to each user account in the
  service provider account.

.. tip::
   To access this information through the *Axivox* portal, click :guilabel:`Users`, then choose a
   user and click :guilabel:`Edit`, and finally click the :guilabel:`SIP Identifiers` tab.

Once all of this information is entered, tap the green :guilabel:`Register` button at the top of the

> screen. *Zoiper* will confirm the information, then show a message saying :guilabel:`Registration

Reviewer 2: It looks like “Lite” was dropped from Zoiper Lite here—intentional, or should we add it back? 🤔

Status: OK` if everything was correct. From here, the *Zoiper Lite* app can now be used to make and
take phone calls using the |VOIP| service.

.. _voip/linphone:

Linphone
========

*Linphone* is a third-party alternative to Odoo |VOIP|. It is a free |VOIP| |SIP| dialer with voice,
video, messaging (direct messages and group chats), and can support conference calls.

To start, download the *Linphone* app from either the `Google's Play Store
<https://play.google.com/store/apps/details?id=org.linphone>`_ or the `Apple App Store
<https://apps.apple.com/us/app/linphone/id360065638>`_. This section will cover setting up
*Linphone* for an iOS device while using :doc:`Axivox <axivox/axivox_config>` as the |VOIP| service

> provider, so the experience may differ for other operating systems or |VOIP| service providers.

Reviewer 2: It’s unclear where Axivox fits into the flow. We say “using Axivox as the VoIP provider,” but never show when or where to obtain and enter the Axivox SIP credentials. Could you: 
1. Add a brief prerequisite (“Create an Axivox SIP account in the Axivox portal and copy the SIP username, password, and domain”), or

2. Link to existing Axivox-setup docs and specify exactly where those credentials are pasted in Linphone?

That context will help readers complete the configuration without guessing.
